test/distributed/_composable/test_checkpoint.py

```diff
@@ -184,17 +174,59 @@ def test_multi_args(self, use_reentrant: bool):
             MultiInputModel(device),
         )
         net2 = deepcopy(net1)
-        checkpoint(net2[0], use_reentrant=use_reentrant)
-        checkpoint(net2[2], use_reentrant=use_reentrant)
+        checkpoint(net2[0])
+        checkpoint(net2[2])
         x1 = torch.randn(20, 100, requires_grad=True)
         x2 = x1.clone()
         net1(x1).sum().backward()
         net2(x2).sum().backward()
         for p1, p2 in zip(net1.parameters(), net2.parameters()):
             self.assertEqual(p1.grad, p2.grad)
 
+    def test_raise_in_forward(self):
+
+        class MyModel(torch.nn.Module):
```

Comment:
nit: For AC specifically, it can be good to think about applying it to a 'module' as opposed to a 'model' because realistically we would never checkpoint the entire model.

Reply:
sounds good, will refactor this into a container